### PR TITLE
add tool use intent metrics

### PR DESCRIPTION
**Pull request by fnan**
Add more tool use metrics to capture if the prediction has the correct tool call intent as the ground truth.

### CODE REVIEW

**Review by ruomingp — COMMENTED** [review]
Could you add PR descriptions?

File: axlearn/open_api/metrics/tool_use_execution.py [hunk 1]
```diff
@@ -424,6 +424,27 @@ def metric_fn(
...
+            stop words in the argument values.
+        - bow_accuracy: Bag-of-words accuracy. Transforms the argument strings in the same way
+            as the lenient matching. But instead of comparing the resulting strings it checks
+            if the words in the ground truth argument values are contained in the predicted
+            argument values.
+        - func_name_accuracy: Accuracy of matching the ground truth tool names.
+        - number_of_expected_tool_calls: Number of expected tool calls.
+        - num_func_call_intents_ground_truth: Number of assistant turns with tool calls
+            in the ground truth.
+        - num_func_call_intents_pred: Number of assistant turns with tool calls in predictions.
```
Comment (ruomingp) [inline axlearn/open_api/metrics/tool_use_execution.py:444]: Please use `num_` vs. `number_of_` consistently.

File: axlearn/open_api/metrics/tool_use_execution.py [hunk 2]
```diff
@@ -424,6 +424,27 @@ def metric_fn(
...
+        - func_intent_recall: Recall of tool calls intent.
+        - func_intent_precision": Precision of tool calls intent.
```
Comment (ruomingp) [inline axlearn/open_api/metrics/tool_use_execution.py:446]:
```suggestion
        - func_intent_precision: Precision of tool calls intent.
```

File: axlearn/open_api/metrics/tool_use_execution.py [hunk 3]
```diff
@@ -424,6 +424,27 @@ def metric_fn(
...
         A dictionary of metrics.
+        - number_of_examples: Total number of examples.
+        - number_of_parsing_errors: Total number of examples with parsing errors.
+        - number_of_generation_errors: Total number of examples with generation errors.
+        - accuracy: Accuracy after standardization of argument values including removing spaces,
+            punctuation, and converting to lowercases. Also apply matching rules if provided.
+        - strict_accuracy: Strict accuracy where argument values exactly matches the
+            ground truth or not.
+        - lenient_accuracy: Accuracy after removing spaces, punctuation and a static list of
+            stop words in the argument values.
```
Comment (ruomingp) [inline axlearn/open_api/metrics/tool_use_execution.py:435]: Point to normalization functions for each type of accuracy?

Comment (fnan) [reply axlearn/open_api/metrics/tool_use_execution.py]: Added.

File: axlearn/open_api/metrics/tool_use_execution.py (hunk 1 above)
Comment (fnan) [reply axlearn/open_api/metrics/tool_use_execution.py]: Done.

File: axlearn/open_api/metrics/tool_use_execution.py (hunk 2 above)
Comment (fnan) [reply axlearn/open_api/metrics/tool_use_execution.py]: Done.

**Review by ruomingp — APPROVED** [review]

**Pull request merged**